### ISSUE TITLE
re-enable Message Filter in AMCL

### ISSUE THREAD

**Issue by mhpanah** [ros-planning/navigation2#171]
currently geometry2 message_filter is not ported to ROS2.  Once the message filter is ported to ROS2 we need to re-enable it in AMCL.  

@gaoethan Would you be able to help us with this?

**Comment by gaoethan** [comment]
Yes. generally speaking, we can help on this. when do you plan to use it ?

**Comment by mkhansen-intel** [comment]
@gaoethan - we're currently using a work-around in the code to make AMCL work without it, so we could use this change as soon as you can make it. How long would it take?

**Comment by gaoethan** [comment]
Okay, I think that it's enough for me to port it for you in one week.

**Comment by mkhansen-intel** [comment]
@gaoethan - 2 weeks is fine if you have other work that is higher priority

**Comment by yechun1** [comment]
Seems the message_filter interface of tf2_ros from geometry2 is not ported, let me have a try to port it.

**Comment by mkhansen-intel** [comment]
Also linking to our issue on this #132

**Comment by gaoethan** [comment]
I've already worked out an available one, please have a try with the following code base:
https://github.com/gaoethan/message_filters.git with branch `master`
https://github.com/gaoethan/geometry2.git with branch `tf2_ros_message_filters`

**Comment by mkhansen-intel** [comment]
When we enable the tf2 message filters, we need to revisit the logging changes made in PR #328. I'm changing some ERRORS and INFO to warnings in that issue to quiet some of the message spew due to the lack fo message filters. If we can enable message filters, these should quiet down anyway, and those log messages can go from DEBUG back to ERROR / INFO.

**Comment by mkhansen-intel** [comment]
Also need to revert the change in #340 when message filters are working

**Comment by gaoethan** [comment]
My porting  (https://github.com/ros2/geometry2/pull/81) to tf2 message filters has already been merged into geometry2, you can try to use it now and please file issue while you find.

**Comment by crdelsey** [comment]
Probably #373 should be backed out when message filter functionality is re-enabled.
